Active Record Validations > conditional validation (#77)
campuscode/rails-guides-pt-BR: pt-BR/active_record_validations.md

válidos. Validações de *model* são a melhor maneira de garantir que só
dados válidos sejam salvos em seu banco de dados. Eles são bancos de dados

> agnóticos, não podem ser contornador por usuários, e são fáceis de manter e

Reviewer: ```suggestion
agnósticos, não podem ser contornados por usuários, e são fáceis de manter e
```

* `update`
* `update!`

As versões *bang* (ex: `save!`) levantam uma exceção se o objeto for
inválido. As versões normais não fazem isso: `save` e `update` retornam `false`,
e `create` retorna o objeto.

### Pulando Validações

Os seguintes métodos pulam validações, e irão salvar o objeto no banco

> de dados independente da sua validez. Eles devem ser usados com cuidado.

Reviewer 2: ```suggestion
de dados independente da sua validade. Eles devem ser usados com cuidado.
```

```

`invalid?` é simplesmente o inverso de `valid?`. Desencadeia suas validações e

> retornam `true` se algum erro for encontrado no objeto, e `false`

Reviewer 2: ```suggestion
retorna `true` se algum erro for encontrado no objeto, e `false`
```
Acredito que seja no singular por se referir ao 'invalid?'

>> person.errors.details[:name] # => [{error: :blank}]
```

O uso de `details` juntamente com validadores é tratado na seção
[Trabalhando com Erros de Validações](#trabalhando-com-erros-de-validações).

> Validation Helpers

Reviewer 2: Acredito que faltou traduzir esse trecho

Person.new.valid?  # => TokenGenerationException: Token can't be blank
```

Validação com Condicional
----------------------

> As vezes fará sentido validar um objeto apenas quando uma determinada condição acontecer.

Reviewer 2: ```suggestion
Às vezes fará sentido validar um objeto apenas quando uma determinada condição for satisfeita.
```
Acho que em português,  o "jargão" usado é satisfazer a condição, mas fiquei na dúvida.

> Você pode fazer isso usando `:if` e `:unless`, que podem ser usandos como um *symbol*,

Reviewer: ```suggestion
Você pode fazer isso usando `:if` e `:unless`, que podem ser usados como um *symbol*,
```

end
```

### Usando uma *Proc* com `:if` e `:unless`

> É possível associar `:if` e `:unless` com um objeto `Proc` que será chamada.

Reviewer: ```suggestion
É possível associar `:if` e `:unless` com um objeto `Proc` que será chamado.
```

O uso de um objeto `Proc` permite escrever uma condição em apenas uma linha
ao invés de em um método separado.

> Esta opção é mais adequada para *one-liners*.

Reviewer 3: Acho que a tradução da frase acima deu contexto suficiente pra não precisar dessa linha mais. O que você acha @IngridRodrigues15 e @campuscode/guia-rails-team ?

Author: Tirei essa linha

end
```

> Todas as validações dentro do bloco `with_options` teram automaticamente passado pela validação  `if: :is_admin?`

Reviewer: ```suggestion
Todas as validações dentro do bloco `with_options` terão automaticamente passado a condição `if: :is_admin?`
```
